### DISCUSSION TITLE
How come the model size increases as the input image number increases?

### DISCUSSION THREAD

**Discussion by KaanDai** [open-edge-platform/anomalib#1491] [Q&A] [answered]
**How come the model size increases as the input image number increases?** I tested with 2 different datasets.

Settings for both tests : 
head model patchcore
input 256x256
backbone model resnet18
test 10% (from_dir)
validation 50% of the test (from_test)
layers 2-3

Test A: 
Total (normal & abnormal) 128 images  --> Onnx model size is almost 30 MB

Test B:
Total (normal & abnormal) 64 images  --> Onnx model size is almost 22 MB

_This difference becomes much more noticeable if you change the backbone model type._

**Comment by abc-125** [comment, marked as answer]
Hello, this is the expected behavior for PatchCore. The model's size depends on the amount of images in the training set. 
Patchcore is not trained in the usual meaning of it, instead it extracts features from the training dataset into the memory bank. More images mean more features extracted, the memory bank is bigger, size of the model increases. If you want to keep the size low, you can try lower `coreset_sampling_ratio.`

**Reply by KaanDai** [reply]
Thank you for your fast response.
